packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1185,7 +1185,7 @@ function commitRootWhenReady(
 ) {
   // TODO: Combine retry throttling with Suspensey commits. Right now they run
   // one after the other.
-  if (finishedWork.subtreeFlags & ShouldSuspendCommit) {
+  if (finishedWork.subtreeFlags & (ShouldSuspendCommit | Visibility)) {
```

Comment:
To make this even more specific, if ShouldSuspendCommit is not present, it should check if both Visibility _and_ MaySuspendCommit are set. If there's no MaySuspendCommit flag it can bail out.

So something like:

```js
const BothShouldSuspendCommitAndVisibility = ShouldSuspendCommit | Visibility;
if (
  subtreeFlags & SoundSuspendCommit ||
  ((subtreeFlags & BothShouldSuspendCommitAndVisibility) === BothShouldSuspendCommitAndVisibility)
```